### PR TITLE
Add custom callback handling

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Description

Add the ability to do custom callback handling. Used to store/process incoming authorization artifacts or other login-time transactions. See EXAMPLES.md for how to use this option.

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

**Comment by joshcanhelp** [comment]
@jfromaniello @hzalaz - Would be curious to get your feedback on this approach. We want a provide a way to do some additional validation/processing/etc on the callback route before we call the user signed in.

**Comment by panva** [comment]
@joshcanhelp 

If this example's going to end up in a README please adjust the `client.userinfo` call to get the TokenSet instance as an argument.

**Comment by joshcanhelp** [comment]
@panva - Better?

https://github.com/auth0/express-openid-connect/pull/37/files#diff-27b714cfcd9ee7241d8db7efa832b94aR154

**Comment by panva** [comment]
> @panva - Better?
> 
> [auth0/express-openid-connect/pull/37/files#diff-27b714cfcd9ee7241d8db7efa832b94aR154](https://github.com/auth0/express-openid-connect/pull/37/files#diff-27b714cfcd9ee7241d8db7efa832b94aR154)

yes!

File: EXAMPLES.md [hunk 1]
```diff
@@ -150,3 +150,28 @@ app.get('/route-that-calls-an-api', async (req, res, next) => {
...
+## 5. Calling userinfo
+
+If your application needs to call the userinfo endpoint for the user's identity, add a `handleCallback` function during initialization that will make this call. To map the incoming claims to the user identity, also add a `getUser` function.
+
+```js
+app.use(auth({
+  handleCallback: async function (req, res, next) {
+    const client = req.openid.client;
+    try {
+      req.session.openidTokens.userinfo = await client.userinfo(req.session.openidTokens);
```
Comment (jfromaniello) [inline EXAMPLES.md]: Why storing the userInfo in the TokenSet instead of just in the session?

**Pull request merged**